pkg/private/tar/build_tar.py

```diff
@@ -64,18 +77,14 @@ def add_file(self, f, destfile, mode=None, ids=None, names=None):
        names: (username, groupname) for the file to set ownership. `f` will be
          copied to `self.directory/destfile` in the layer.
     """
-    dest = destfile.lstrip('/')  # Remove leading slashes
-    if self.directory and self.directory != '/':
-      dest = self.directory.lstrip('/') + '/' + dest
+    dest = self.normalize_path(destfile)
```

Comment:
Do we remain the same behavior if we change `dest` from related path to possibly absolute path?

Reply:
Yes, this is fine.   'dest' is just the path we put in the tarfile. It should be relative all the time, but someone
might want to set the package prefix to something absolute like '/etc'. That will still give a valid tar file.

Of course, doing that is almost always a bad idea. All reasonable implementations will strip the leading / on untar.